src/Orleans.Core/Async/UnobservedExceptionsHandler.cs

```diff
@@ -6,17 +6,18 @@ namespace Orleans
 {
     using Orleans.Runtime;
 
-    internal class UnobservedExceptionsHandlerClass
+    internal class UnobservedExceptionsHandler : IDisposable
     {
-        private static readonly Object lockObject = new Object();
-        private static ILogger logger;
-        private static UnobservedExceptionDelegate unobservedExceptionHandler;
-        private static readonly bool alreadySubscribedToTplEvent = false;
+        private readonly Object lockObject = new Object();
+        private readonly ILogger logger;
+        private UnobservedExceptionDelegate unobservedExceptionHandler;
+        private bool alreadySubscribedToTplEvent = false;
```

Comment:
no reason for this field any more, since it's only subscribed during construction anyway